templates/CRM/Contribute/Form/Search/Common.tpl

```diff
@@ -57,13 +57,15 @@
   <td>
     {$form.contribution_or_softcredits.label} <br />
     {$form.contribution_or_softcredits.html}
```

Comment:
Adding a <br /> at the end of line 59 fixes the display problem which occurs when the Soft Credit Type widget is un-hidden (e.g. when selecting some other than 'Contributions Only' in 'Contributions OR Soft Credits' dropdown). Puts the Soft Credit Type select2 on the next line which I think is fine.